File: backend/src/config/storage.ts
import { mkdirSync, existsSync } from "node:fs";
import { resolve } from "node:path";
import { env } from "./env.js";

const uploadRoot = resolve(process.cwd(), env.UPLOAD_DIR);

const SUBDIRS = [
  "mtc",
  "invoices",
  "delivery-notes",
  "qc-reports",
  "inspection-photos",
  "installation-photos",
  "complaint-photos",
  "misc",
] as const;

export type UploadCategory = (typeof SUBDIRS)[number];

export function ensureUploadDirectories(): string {
  if (!existsSync(uploadRoot)) {
    mkdirSync(uploadRoot, { recursive: true });
  }

  for (const subdir of SUBDIRS) {
    const path = resolve(uploadRoot, subdir);
    if (!existsSync(path)) {
      mkdirSync(path, { recursive: true });
    }
  }

  return uploadRoot;
}

export function getUploadPath(category: UploadCategory, filename: string): string {
  return resolve(uploadRoot, category, filename);
}

export { uploadRoot, SUBDIRS };

File: backend-go/internal/config/storage.go
package config

import (
	"os"
	"path/filepath"
)

var UploadSubdirs = []string{
	"mtc",
	"invoices",
	"delivery-notes",
	"qc-reports",
	"inspection-photos",
	"installation-photos",
	"complaint-photos",
	"misc",
}

func EnsureUploadDirectories(uploadDir string) (string, error) {
	root, err := filepath.Abs(uploadDir)
	if err != nil {
		return "", err
	}
	if err := os.MkdirAll(root, 0o755); err != nil {
		return "", err
	}
	for _, sub := range UploadSubdirs {
		if err := os.MkdirAll(filepath.Join(root, sub), 0o755); err != nil {
			return "", err
		}
	}
	return root, nil
}
